src/delta.f90

```diff
@@ -644,6 +658,77 @@ subroutine fill_tetrahedra_3d(tetra, evals, tetra_evals)
        end do
     end do
   end subroutine fill_tetrahedra_3d
+  
+  
+subroutine fermi_window_adjusted_fill_tetrahedra_3d(wann, crys, wave_vector_mesh, tetra, evals, tetra_evals)
+
+    type(wannier), intent(in) :: wann
+    type(crystal), intent(in) :: crys
+    integer(i64), intent(in) :: tetra(:,:)
+    real(r64), intent(in) :: evals(:,:)
+    integer(i64), intent(in) :: wave_vector_mesh(1, 3)
+    
+    real(r64), allocatable, intent(out) :: tetra_evals(:,:,:)
+
+    integer(i64) :: iv, it, ib, numbands, aux, numtetra
+    integer(i64) :: kpoint(3)
+    
+    real(r64), allocatable :: energies(:,:)
+    real(r64) :: kvecs(1,3)
+
+    ! Calculate dimensions
+    numtetra = size(tetra, 1)
+    numbands = size(evals, 2)
+
+    print *, "Number of tetrahedra (numtetra) =", numtetra
+    print *, "Number of bands (numbands) =", numbands
+
+	  print *, "tetra shape:", shape(tetra)
+	  print *, "evals shape:", shape(evals)
+	  print *, "tetra_evals shape:", shape(tetra_evals)
+
+    ! Allocate arrays with correct dimensions
+   allocate(tetra_evals(numtetra, numbands, 4))
+   allocate(energies(1, wann%numwannbands))
+
+    ! Initialize tetra_evals and energies
+    tetra_evals(:,:,:) = 0.0_r64
```

Comment:
It is preferable to to write `tetra_evals(:, :, :)` instead of `tetra_evals(:,:,:)`. This style rule will be strictly enforced in the near future.